File: 백준/Silver/11723. 집합/집합.cc
#include <bits/stdc++.h>
using namespace std;

int M, num, S[30];
string s;

int main(void) {
    ios_base::sync_with_stdio(0); cin.tie(0); cout.tie(0);

    cin >> M;

    while(M--) {
        cin >> s;

        if(s == "add") {
            cin >> num;
            S[num] = 1;
        }
        if(s == "remove") {
            cin >> num;
            S[num] = 0;
        }
        if(s == "check") {
            cin >> num;
            cout << S[num] << '\n';
        }
        if(s == "toggle") {
            cin >> num;
            if(S[num]) S[num] = 0;
            else S[num] = 1;
        }
        if(s == "all") {
            for(int i = 1; i <= 20; i++) S[i] = 1;
        }
        if(s == "empty") {
            for(int i = 1; i <= 20; i++) S[i] = 0;
        }
    }
}

File: 백준/Silver/11723. 집합/집합.kt
import java.io.BufferedReader
import java.io.BufferedWriter
import java.io.InputStreamReader
import java.io.OutputStreamWriter

fun main() {
    val br = BufferedReader(InputStreamReader(System.`in`));
    val bw = BufferedWriter(OutputStreamWriter(System.out));

    val N = br.readLine().toInt();
    val S = BooleanArray(21);

    (1..N).forEach {
        val op = br.readLine().split(" ");

        if(op[0]=="add") {
            val num = op[1].toInt();
            S[num] = true;
        }
        if(op[0]=="remove") {
            val num = op[1].toInt();
            S[num] = false;
        }
        if(op[0]=="check") {
            val num = op[1].toInt();
            if(S[num]) bw.write(1.toString());
            else bw.write(0.toString());
            bw.newLine();
        }
        if(op[0]=="toggle") {
            val num = op[1].toInt();
            if(S[num]) S[num] = false;
            else S[num] = true;
        }
        if(op[0]=="all") {
            for(i in 1..20) {
                S[i] = true;
            }
        }
        if(op[0]=="empty") {
            for(i in 1..20) {
                S[i] = false;
            }
        }
    }

    br.close();
    bw.flush();
    bw.close();
}